Sources/VLCTranscoder.m

```diff
@@ -0,0 +1,142 @@
+}
+
+- (BOOL)muxSubtitleFile:(NSString *)srtPath toMp4File:(NSString *)mp4Path outputPath:(NSString *)outputPath
+{
+    libvlc_media_t* p_media = libvlc_media_new_path([[VLCLibrary sharedLibrary] instance], [mp4Path UTF8String]);
+    if (p_media == NULL) {
+        NSAssert(0, @"p_media wasn't allocated");
+        return NO;
+    }
+    NSString *transcodingOptions = [NSString stringWithFormat:@":sout=#transcode{venc={module=avcodec{codec=h264_videotoolbox}, vcodec=h264},venc={module=vpx{quality-mode=2},vcodec=VP80},acodec=mpga,ab=128,channels=2,samplerate=44100,soverlay}:file{dst='%@',mux=mkv}", outputPath];
+    libvlc_media_add_option(p_media, [[NSString stringWithFormat:@"--sub-file=%@", srtPath] UTF8String]);
+    libvlc_media_add_option(p_media, [transcodingOptions UTF8String]);
+
+    _p_mp = libvlc_media_player_new_from_media( p_media );
+    if (_p_mp == NULL) {
+        NSAssert(0, @"_p_mp wasn't allocated");
+        return NO;
+    }
+
+    [self registerObserversForMuxWithPlayer:_p_mp];
+    BOOL canPlay = libvlc_media_player_play( _p_mp ) == 0;
+    if (!canPlay) {
+        NSAssert(0, @"playback failed");
+        [self unregisterObserversForMuxWithPlayer:_p_mp];
+        return NO;
+    }
+    return canPlay;
+}
+
+- (void)registerObserversForMuxWithPlayer:(libvlc_media_player_t *)player
+{
+    __block libvlc_event_manager_t * p_em = libvlc_media_player_event_manager(player);
+    if (!p_em)
+        return;
+    dispatch_sync(_libVLCTranscoderQueue,^{
+        libvlc_event_attach(p_em, libvlc_MediaPlayerPaused,
+                            HandleMuxMediaInstanceStateChanged, (__bridge void *)(self));
+        libvlc_event_attach(p_em, libvlc_MediaPlayerStopped,
+                            HandleMuxMediaInstanceStateChanged, (__bridge void *)(self));
+        libvlc_event_attach(p_em, libvlc_MediaPlayerEncounteredError,
+                            HandleMuxMediaInstanceStateChanged, (__bridge void *)(self));
+    });
+}
+
+- (void)unregisterObserversForMuxWithPlayer:(libvlc_media_player_t *)player
+{
+    libvlc_event_manager_t * p_em = libvlc_media_player_event_manager(player);
+    if (!p_em)
+        return;
+    libvlc_event_detach(p_em, libvlc_MediaPlayerStopped,
+                        HandleMuxMediaInstanceStateChanged, (__bridge void *)(self));
+    libvlc_event_detach(p_em, libvlc_MediaPlayerPaused,
+                        HandleMuxMediaInstanceStateChanged, (__bridge void *)(self));
+    libvlc_event_detach(p_em, libvlc_MediaPlayerEncounteredError,
+                        HandleMuxMediaInstanceStateChanged, (__bridge void *)(self));
+}
+
+- (void)mediaPlayerStateChangeForMux:(NSNumber *)newState
+{
+    if (_p_mp) {
+        [self unregisterObserversForMuxWithPlayer:_p_mp];
+        libvlc_media_player_stop( _p_mp );
+        if (self.delegate && [self.delegate respondsToSelector:@selector(transcoderFinishedSucessfully:)]) {
+            BOOL success = [newState isEqualToNumber: @(VLCMediaPlayerStateError)];
```

Comment:
`success` is the opposite of the intended meaning here, even though one line down it is negated this is a bit confusing when reading the code, better immediately negate the result of the `isEqualToNumber` method here when assigning and remove the negation below.

Reply:
very good point